src/Servers/Kestrel/Transport.Sockets/src/SocketConnectionContextFactory.cs

```diff
@@ -0,0 +1,68 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers;
+using System.IO.Pipelines;
+using System.Net.Sockets;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets.Internal;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Sockets
+{
+    /// <summary>
+    /// A factory for socket based connections contexts.
+    /// </summary>
+    internal sealed class SocketConnectionContextFactory : ISocketConnectionContextFactory, IAsyncDisposable
```

Comment:
Should be `public`, yes?

Reply:
Why?

Reply:
I might be misunderstanding, but I thought that's what [the public API entry here](https://github.com/dotnet/aspnetcore/pull/34769/files#diff-8b15fbe07f965adfec63a9c4ca239ff43dd1310dcd9d6f8bd69b49a78ca76fb9R9) indicated.

I also don't quite follow how'd you get a handle on this, to use separately from `SocketTransportFactory`.

Reply:
Dependency injection

Reply:
So, call `UseSockets()`, grab this factory, and then remove both the registered `SocketTransportFactory` and this factory?  Feels a little weird to register both, when presumably a custom transport is almost always gonna be registered.

Also seems kind of brittle if `UseSockets()` registers anything beyond those two in the future.